lib/bigtable/filter.js

```diff
@@ -0,0 +1,903 @@
+ * //-
+ * // Using the following filter, we would retrieve the `tjefferson` and
+ * // `gwashington` columns.
+ * //-
+ * var filter = [
+ *   {
+ *     column: /[a-z]+on$/
+ *   }
+ * ];
+ *
+ * //-
+ * // You can also provide a string (optionally containing regexp characters)
+ * // for simple column filters.
+ * //-
+ * var filter = [
+ *   {
+ *     column: 'gwashington'
+ *   }
+ * ];
+ *
+ * //-
+ * // Or you can provide an array of strings if you wish to match against
+ * // multiple columns.
+ * //-
+ * var filter = [
+ *   {
+ *     column: [
+ *       'gwashington',
+ *       'tjefferson'
+ *     ]
+ *   }
+ * ];
+ *
+ * //-
+ * // If you wish to use additional column filters, consider using the following
+ * // syntax.
+ * //-
+ * var filter = [
+ *   {
+ *     column: {
+ *       name: 'gwashington'
+ *     }
+ *   }
+ * ];
+ *
+ *
+ * //-
+ * // <h4>Column Cell Limits</h4>
+ * //
+ * // Matches only the most recent number of versions within each column. For
+ * // example, if the `versions` is set to 2, this filter would only match
+ * // columns updated at the two most recent timestamps.
+ * //
+ * // If duplicate cells are present, as is possible when using an
+ * // {module:bigtable/filter#interleave} filter, each copy of the cell is
+ * // counted separately.
+ * //-
+ * var filter = [
+ *   {
+ *     column: {
+ *       cellLimit: 2
+ *     }
+ *   }
+ * ];
+ *
+ * //-
+ * // <h4>Column Ranges</h4>
+ * //
+ * // Specifies a contiguous range of columns within a single column family.
+ * // The range spans from <column_family>:<start_qualifier> to
+ * // <column_family>:<end_qualifier>, where both bounds can be either
+ * // inclusive or exclusive. By default both are inclusive.
+ * //
+ * // When the `start` bound is omitted it is interpreted as an empty string.
+ * // When the `end` bound is omitted it is interpreted as Infinity.
+ * //-
+ * var filter = [
+ *   {
+ *     column: {
+ *       family: 'follows',
+ *       start: 'gwashington',
+ *       end: 'tjefferson'
+ *     }
+ *   }
+ * ];
+ *
+ * //-
+ * // By default, both the `start` and `end` bounds are inclusive. You can
+ * // override these by providing an object explicity stating whether or not it
+ * // is `inclusive`.
+ * //-
+ * var filter = [
+ *   {
+ *     column: {
+ *       family: 'follows',
+ *       start: {
+ *         value: 'gwashington',
+ *         inclusive: false
+ *       },
+ *       end: {
+ *         value: 'jadams',
+ *         inclusive: false
+ *       }
+ *     }
+ *   }
+ * ];
+ */
+Filter.prototype.column = function(column) {
+  if (!is.object(column)) {
+    column = {
+      name: column
+    };
+  }
+
+  if (column.name) {
+    var name = Filter.convertToRegExpString(column.name);
+
+    name = Mutation.convertToBytes(name);
+    this.set('columnQualifierRegexFilter', name);
+  }
+
+  if (is.number(column.cellLimit)) {
+    this.set('cellsPerColumnLimitFilter', column.cellLimit);
+  }
+
+  if (column.start || column.end) {
```

Comment:
Could either value ever be caught as falsy (i.e. `0`) and end up bypassing this?